src/catalog/default/default_functions.cpp

```diff
@@ -62,6 +62,7 @@ static const DefaultMacro internal_macros[] = {
 	{"pg_catalog", "pg_get_expr", {"pg_node_tree", "relation_oid", nullptr}, "pg_node_tree"},
 	{"pg_catalog", "format_pg_type", {"logical_type", "type_name", nullptr}, "case when logical_type='FLOAT' then 'real' when logical_type='DOUBLE' then 'double precision' when logical_type='DECIMAL' then 'numeric' when logical_type='ENUM' then lower(type_name) when logical_type='VARCHAR' then 'character varying' when logical_type='BLOB' then 'bytea' when logical_type='TIMESTAMP' then 'timestamp without time zone' when logical_type='TIME' then 'time without time zone' else lower(logical_type) end"},
 	{"pg_catalog", "format_type", {"type_oid", "typemod", nullptr}, "(select format_pg_type(logical_type, type_name) from duckdb_types() t where t.type_oid=type_oid) || case when typemod>0 then concat('(', typemod//1000, ',', typemod%1000, ')') else '' end"},
+	{"pg_catalog", "map_to_pg_oid", {"oid", "logical_type", nullptr}, "case logical_type when 'ENUM' then oid else case oid when 10 then 16 when 12 then 21 when 13 then 23 when 14 then 20 when 15 then 1082 when 16 then 1083 when 19 then 1114 when 22 then 700 when 23 then 701 when 25 then 1043 when 26 then 17 when 27 then 1186 when 32 then 1184 when 34 then 1266 when 36 then 1560 when 54 then 2950 else null end end"}, // map duckdb_oid to pg_oid. If no corresponding type, return null 
```

Comment:
This is not robust - the oids in DuckDB are not guaranteed to be constants and might change over time. Could you rework this to use the names of the types instead?